src/main.cpp

```diff
@@ -4,8 +4,30 @@
 #include <CLI/CLI.hpp>
 #include <spdlog/sinks/stdout_color_sinks.h>
 
-int main(int argc, char **argv) {
+CLI::Option *SetKeyValueOptions(CLI::App &App, const std::string &Name,
+                                const std::string &Description, bool Defaulted,
+                                const CLI::callback_t &Fun) {
+  CLI::Option *Opt = App.add_option(Name, Fun, Description, Defaulted);
+  const auto RequireEvenNumberOfPairs = -2;
+  Opt->type_name("KEY VALUE");
+  Opt->type_size(RequireEvenNumberOfPairs);
+  return Opt;
+}
 
+CLI::Option *addKafkaOption(CLI::App &App, std::string const &Name,
+                            std::map<std::string, std::string> &ConfigMap,
+                            std::string const &Description,
+                            bool Defaulted = false) {
+  CLI::callback_t Fun = [&ConfigMap](CLI::results_t Results) {
+    for (size_t i = 0; i < Results.size() / 2; i++) {
```

Comment:
Can you add a comment here to say why you are iterating through like this? it's not immediately obvious

Reply:
Done